helix-term/src/commands.rs

```diff
@@ -3916,18 +3958,30 @@ pub mod insert {
                     .map_or(false, |pair| pair.open == prev && pair.close == curr);
 
                 let local_offs = if on_auto_pair {
```

Comment:
We can add an extra block to special case comment continuing here:


```rust
let local_offs = if let Some(token) = continue_comment_token {
    // insert the line ending, then indent then token then ' '
} else if on_auto_pair() {
    // unchanged
} else {
    // unchanged
}
```

and switch `on_auto_pair` to be a closure that takes no arguments - that way we avoid computing it if we're continuing a comment